test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +79,77 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-unanalyzed-fields [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (assert (pos? ;; don't let ourselves be fooled if the test passes because the table is
+                  ;; totally broken or has no fields. Make sure we actually test something
+             (db/count Field :table_id table-id)))
+    (db/count Field :last_analyzed nil :table_id table-id)))
+
+(defn latest-sync-time [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (db/select-field :last_analyzed Field :table_id table-id)))
```

Comment:
(nvm I guess you sort of do want the actual sync time. disregard the last comment) you should still maybe change the call to `select-one-field` or at the very least change the function name so it matches what it does